### DISCUSSION TITLE
Appending old version to new version

### DISCUSSION THREAD

**Discussion by phyzical** [renovatebot/renovate#31350] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github action

### Please tell us more about your question or problem

Hey

Basically i want to know if its currently possible to append the old version after the new version

what im trying to achieve is an array that automatically gets additional items added via renovate i.e 

```
// SOME_RANDOM_MATCH_STRING
version = [
   "1.0.0",
   "0.9.0",
   "0.8.0"
]
```

So that on an update i would see

```
// SOME_RANDOM_MATCH_STRING
version = [
   "1.1.0",
   "1.0.0",
   "0.9.0",
   "0.8.0"
]
```

i know currently i could make it do the following, 
```
// SOME_RANDOM_MATCH_STRING
version = [
   "1.1.0",
   "0.9.0",
   "0.8.0"
]
```

but what im trying to achieve is a list that can then be parsed and provided to spin off a series of github matrix builds (ruby base image builds)

Thanks!

**Comment by rarkins** [comment]
Unfortunately Renovate doesn't have any "manipulate a matrix of versions" capability currently, although it's an idea which probably applies across a few ecosystems

**Comment by phyzical** [comment]
@rarkins thanks for the reply i guess i was hoping we. stored `currentValue` and `newValue` (Or `oldValue`) to manipulate the final output.

Looking at a few datasource implementations is it a correct assumption that DS wise expects that the `currentValue` is simply replaced as part of the flow and reinserted somewhere towards the end?

Just trying to make sure there isnt a way to make this happen with a clever regex 

If so in theory, we just need to tack on `oldValue = currentValue` before any replacements and allow for it to be part of the replacement keywords correct?

**Reply by rarkins** [reply]
If you're playing around with custom managers and possibly also datasources then maybe anything is possible, but otherwise this is not a core capability of Renovate which I can advise you on
